drivers/adc/ad7606/ad7606.c

```diff
@@ -595,6 +610,65 @@ int32_t ad7606_spi_data_read(struct ad7606_dev *dev, uint32_t *data)
 	return ret;
 }
 
+/***************************************************************************//**
+ * @brief Read multiple samples using the AXI SPI engin code
+ *
+ * @param dev        - The device structure.
+ * @param data       - Pointer to location of buffer where to store the data.
+ * @param samples    - Number of samples to read
+ *
+ * @return ret - return code.
+ *         Example: -EIO - SPI communication error.
+ *                  -ETIME - Timeout while waiting for the BUSY signal.
+ *                  -EBADMSG - CRC computation mismatch.
+ *                  0 - No errors encountered.
+*******************************************************************************/
+static int32_t ad7606_read_raw_data_spi_engine(struct ad7606_dev *dev,
+		uint32_t *buf, uint32_t samples)
+{
+	struct ad7606_axi_dev *axi = &dev->axi_dev;
+	int32_t ret;
+	uint32_t commands_data[2] = {0x00, 0x00};
+	struct spi_engine_offload_message msg = {};
+	const uint8_t bits = ad7606_chip_info_tbl[dev->device_id].bits;
+	uint32_t spi_eng_msg_cmds[3] = {
+		CS_LOW,
+		READ(2),
+		CS_HIGH,
+	};
+
+	no_os_axi_io_write(axi->core_baseaddr, AD7606_CORE_RESET, 0x01);
+	ret = no_os_pwm_enable(axi->trigger_pwm_desc);
+	if (ret != 0)
+		return ret;
+
+	dev->spi_desc->mode = NO_OS_SPI_MODE_2;
+	spi_engine_set_speed(dev->spi_desc, dev->spi_desc->max_speed_hz);
+	spi_engine_set_transfer_width(dev->spi_desc, bits);
+
+	ret = spi_engine_offload_init(dev->spi_desc, &axi->offload_init_param);
+	if (ret != 0)
+		return ret;
+
+	msg.commands_data = commands_data;
+	msg.commands = spi_eng_msg_cmds;
+	msg.no_commands = NO_OS_ARRAY_SIZE(spi_eng_msg_cmds);
+	msg.rx_addr = (uint32_t)buf;
+
+	no_os_axi_io_write(axi->core_baseaddr, AD7606_CORE_RESET, 0x00);
```

Comment:
Maybe defines could be used instead of the values written for resetting the core.